test/conftest.py

```diff
@@ -35,16 +35,39 @@ def pytest_addoption(parser):
 def pytest_generate_tests(metafunc):
     """Parameterize tests based on image names"""
 
-    images = metafunc.config.getoption("image")
+    images = metafunc.config.getoption("image").copy()
     for image_dir in metafunc.config.getoption("image_dir"):
-        images = images + [
-            os.path.join(image_dir, dir) for dir in os.listdir(image_dir)
+        images += [
+            os.path.join(image_dir, dir_name) for dir_name in os.listdir(image_dir)
         ]
 
-    # tests with "image" parameter are run against images
+    # tests with "image" parameter are run against image
     if "image" in metafunc.fixturenames:
+        filtered_images = []
+        ids = []
+        for image in images:
+            image_base = os.path.basename(image)
+            test_name = metafunc.definition.originalname
+            if test_name.startswith("test_windows_") and not image_base.startswith(
+                "win-"
```

Comment:
The discrepancy between windows and win is a bit jarring (and also makes me think it's the only thing stopping this being put in tight loop for the OSes, rather than a switch statement.